node/constructors.go

```diff
@@ -976,90 +964,90 @@ func (n *Node) startPrometheusServer(addr string) *http.Server {
 }
 
 // Switch returns the Node's Switch.
-func (n *Node) Switch() *p2p.Switch {
+func (n *nodeImpl) Switch() *p2p.Switch {
 	return n.sw
 }
 
 // BlockStore returns the Node's BlockStore.
-func (n *Node) BlockStore() *store.BlockStore {
+func (n *nodeImpl) BlockStore() *store.BlockStore {
 	return n.blockStore
 }
 
 // ConsensusState returns the Node's ConsensusState.
-func (n *Node) ConsensusState() *cs.State {
+func (n *nodeImpl) ConsensusState() *cs.State {
 	return n.consensusState
 }
 
 // ConsensusReactor returns the Node's ConsensusReactor.
-func (n *Node) ConsensusReactor() *cs.Reactor {
+func (n *nodeImpl) ConsensusReactor() *cs.Reactor {
 	return n.consensusReactor
 }
 
 // MempoolReactor returns the Node's mempool reactor.
-func (n *Node) MempoolReactor() service.Service {
+func (n *nodeImpl) MempoolReactor() service.Service {
 	return n.mempoolReactor
 }
 
 // Mempool returns the Node's mempool.
-func (n *Node) Mempool() mempool.Mempool {
+func (n *nodeImpl) Mempool() mempool.Mempool {
 	return n.mempool
 }
 
 // PEXReactor returns the Node's PEXReactor. It returns nil if PEX is disabled.
-func (n *Node) PEXReactor() *pex.Reactor {
+func (n *nodeImpl) PEXReactor() *pex.Reactor {
 	return n.pexReactor
 }
 
 // EvidencePool returns the Node's EvidencePool.
-func (n *Node) EvidencePool() *evidence.Pool {
+func (n *nodeImpl) EvidencePool() *evidence.Pool {
 	return n.evidencePool
 }
 
 // EventBus returns the Node's EventBus.
-func (n *Node) EventBus() *types.EventBus {
+func (n *nodeImpl) EventBus() *types.EventBus {
 	return n.eventBus
 }
 
 // PrivValidator returns the Node's PrivValidator.
 // XXX: for convenience only!
-func (n *Node) PrivValidator() types.PrivValidator {
+func (n *nodeImpl) PrivValidator() types.PrivValidator {
 	return n.privValidator
 }
```

Comment:
What's the reason for keeping all of these if they can't be accessed (I don't think we use them at all in this repo)

Reply:
just trying to keep the diff really simple/rote, and to avoid impacting the tests, some of which uses these accessors, and I don't think they're hurting anything by staying. I think we can definitely do more clean up later, but I'd like to avoid doing anything radical here.

Reply:
ok but let's not forget :)

Reply:
(you can access them if you typecast them to a different interface that has these methods, and some of the RPC tests do this.)